mesos_master/datadog_checks/mesos_master/mesos_master.py

```diff
@@ -178,7 +178,21 @@ def _get_json(self, url, timeout, verify=True, tags=None):
         return r.json()
 
     def _get_master_state(self, url, timeout, verify, tags):
-        return self._get_json(url + '/state.json', timeout, verify, tags)
+        try:
+            # version < 1.8.0
+            endpoint = '/state.json'
+            master_state = self._get_json(url + endpoint, timeout, verify, tags)
+            if master_state is not None:
```

Comment:
This part is duplicated in the try and the except block. You can put that just before the return statement for the same effect. Also, not sure to understand why this part had to be removed from the `_check_leadership` method